src/AnalyticsEventReporter.ts

```diff
@@ -34,10 +34,27 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
   with(payload: EventPayload): AnalyticsEventService {
     const currentPayload =
       this.payload === undefined ? payload : merge(this.payload, payload);
+    if (this.config.debug) {
+      console.log('[DEBUG] AnalyticsConfig Object:\n', this.config);
+      console.log(
+        '[DEBUG] Payload following call to `with`:\n',
+        currentPayload
+      );
+    }
     return new AnalyticsEventReporter(this.config, currentPayload);
   }
 
   public async report(newPayload?: EventPayload): Promise<string> {
+    if (this.config.debug) {
+      console.log(
```

Comment:
I feel like the original payload vs the newPlayload doesn't matter. Just print what would have been sent in the request.